test/Chainweb/Test/Pact/ModuleCacheOnRestart.hs

```diff
@@ -237,28 +236,19 @@ snapshotCache iomcache initCache = do
   mcache <- iomcache
   modifyMVar_ mcache (const (pure initCache))
 
-
 withPact'
     :: IO TestBlockDb
-    -> IO FilePath
+    -> IO SQLiteEnv
     -> IO (MVar ModuleInitCache)
     -> CacheTest RocksDbCas
-    -> (Assertion -> TestTree)
-    -> TestTree
-withPact' bdbio iodir r ctest toTestTree =
-    withResource startPact stopPact go
+    -> Assertion
+withPact' bdbio ioSqlEnv r (ps, cacheTest) = do
```

Comment:
Moving the resource allocation out of `withPact'` allows to simplify the return type from `TestTree` to `Assertion`.